TaskWebApp/App_Start/Startup.Auth.cs

```diff
@@ -141,18 +157,38 @@ private async Task OnAuthorizationCodeReceived(AuthorizationCodeReceivedNotifica
             // Extract the code from the response notification
             var code = notification.Code;
 
-            string signedInUserID = notification.AuthenticationTicket.Identity.FindFirst(ClaimTypes.NameIdentifier).Value;
-            TokenCache userTokenCache = new MSALSessionCache(signedInUserID, notification.OwinContext.Environment["System.Web.HttpContextBase"] as HttpContextBase).GetMsalCacheInstance();
-            ConfidentialClientApplication cca = new ConfidentialClientApplication(ClientId, Authority, RedirectUri, new ClientCredential(ClientSecret), userTokenCache, null);
+            string signedInUserID = notification.JwtSecurityToken.Subject; //notification.AuthenticationTicket.Identity.FindFirst(Startup.ClaimsSubject).Value;
+            ConfidentialClientApplication cca = GetConfidential();
+            var httpContextBase = notification.OwinContext.Environment["System.Web.HttpContextBase"] as HttpContextBase;
+            HttpContext httpContext =  httpContextBase.ApplicationInstance.Context;
+            TokenCacheHelper.EnablePersistence(cca.UserTokenCache);
+            
             try
             {
-                AuthenticationResult result = await cca.AcquireTokenByAuthorizationCodeAsync(code, Scopes);
+                AuthenticationResult result = await cca.AcquireTokenByAuthorizationCode(new List<string>(Scopes), code)
```

Comment:
new List&th;string&gt; is not needed. AcquireTokenByAuthorizationCode takes IEnumerable&th;string&gt;

Reply:
Tiago, please resolve this as well.

---
In reply to: [276995551](https://github.com/Azure-Samples/active-directory-b2c-dotnet-webapp-and-webapi/pull/54#discussion_r276995551) [](ancestors = 276995551)